### DISCUSSION TITLE
Force flow line to always display ?

### DISCUSSION THREAD

**Discussion by vrman** [visgl/flowmap.gl#29] [Q&A] [answered]
when start and end point go out of screen the flow line between them is hide too.
just want to know is that possible to force flow line to always display even if those points go out of screen ?

**Comment by ilyabo** [comment, marked as answer]
There's no prop for that at the moment in `FlowmapLayer`. The adaptive filtering is how the layer makes sure that  you can explore smaller regions in detail when you zoom in, that not too much noise is rendered, and the rendering performance is good.

You can however use the [`FlowLinesLayer`](https://github.com/visgl/flowmap.gl/blob/main/packages/layers/src/FlowLinesLayer/FlowLinesLayer.ts) directly, it will render everything you pass to it.

**Comment by vrman** [comment]
thanks , can you please give me an example of how should be data structure for FlowLinesLayer ? it has getSourcePosition and getTargetPosition , how can set direction for each line  ?

**Reply by ilyabo** [reply]
Sorry, I don't have an example at hand, but I think it should be more or less readable from [this type definition](https://github.com/visgl/flowmap.gl/blob/upgrade-deps/packages/layers/src/FlowLinesLayer/FlowLinesLayer.ts#L15)

```
export interface Props<F> extends LayerProps {
  id: string;
  opacity?: number;
  pickable?: boolean;
  updateTriggers?: {[key: string]: Record<string, unknown>};
  data: F[] | FlowLinesLayerAttributes;
  drawOutline: boolean;
  outlineColor?: RGBA;
  outlineThickness?: number;
  thicknessUnit?: number;
  getSourcePosition?: (d: F) => [number, number];
  getTargetPosition?: (d: F) => [number, number];
  getColor?: (d: F) => RGBA;
  getThickness?: (d: F) => number;
  getPickable?: (d: F, {index}: {index: number}) => number; // >= 1.0 -> true
  getEndpointOffsets?: (d: F) => [number, number];
}
```

**Reply by ilyabo** [reply]
Concerning `getSourcePosition` and `getTargetPosition`, the direction is always from source to target, so you need two separate data items to describe the flows A->B and B->A
